gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -151,7 +156,20 @@ void onSignedOut() {
 
   private void onSignedIn() {
     hideProgressDialog();
-    navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+
+    /* Checking whether the user have accepted Terms & Conditions. */
+    if (localValueStore.isTermsAccepted()) {
+      navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+    } else {
+
+      /*  Checking here whether the user is coming from SignIn Fragment or StartUp Fragment
+       based on that redirecting user. */
+      if (signInProgressDialogVisibility.getValue() == null) {
+        navigator.navigate(StartupFragmentDirections.proceedToTermsScreen());
+      } else {
+        navigator.navigate(SignInFragmentDirections.proceedToTermsScreen());
+      }
+    }
```

Comment:
Prefer early return to deep if else ladders. Try something like 
```
/* Checking whether the user have accepted Terms & Conditions. */
if (localValueStore.isTermsAccepted()) {
      navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
      return;
}

/*  Checking here whether the user is coming from SignIn Fragment or StartUp Fragment
       based on that redirecting user. */
if (signInProgressDialogVisibility.getValue() == null) {
  navigator.navigate(StartupFragmentDirections.proceedToTermsScreen());
  return;
} 

navigator.navigate(SignInFragmentDirections.proceedToTermsScreen());
```

Reply:
this workflow have been changed.

Reply:
workflow changed